src/pudl/etl/ferceqr_deployment.py

```diff
@@ -0,0 +1,145 @@
+from pudl.logging_helpers import get_logger
+from pudl.settings import ferceqr_year_quarters
+from pudl.workspace.setup import PudlPaths
+
+logger = get_logger(__name__)
+
+
+ferceqr_sensor_status = (
+    dg.DefaultSensorStatus.RUNNING
+    if os.getenv("FERCEQR_BUILD", None)
+    else dg.DefaultSensorStatus.STOPPED
+)
+
+FERCEQR_TRANSFORM_ASSETS = [
+    "core_ferceqr__contracts",
+    "core_ferceqr__transactions",
+    "core_ferceqr__quarterly_identity",
+    "core_ferceqr__quarterly_index_pub",
+]
+
+
+def _notify_slack_deployments_channel(
+    message: str, attached_file_path: str | None = None
+):
+    """Send string message to PUDL deployments channel."""
+    client = WebClient(token=os.environ["SLACK_TOKEN"])
+    channel = "C03FHB9N0PQ"
+    if attached_file_path is not None:
+        client.files_upload_v2(
+            channel=channel,
+            file=attached_file_path,
+            title=f"{os.environ['BUILD_ID']} Status",
+            initial_comment=message,
+        )
+    client.chat_postMessage(
+        channel=channel,
+        text=message,
+    )
+
+
+def _write_status_file(status: Literal["SUCCESS", "FAILURE"]):
+    """To notify the top level build script that the job is complete, create a file in the output directory."""
+    (PudlPaths().output_dir / status).touch()
+
+
+@dg.asset
+def deploy_ferceqr():
+    """Publish EQR outputs to cloud storage."""
+    output_location = "gs://builds.catalyst.coop/ferceqr"
+    # Copy parquet files to GCS
+    logger.info("Build successful, copying ferceqr data to GCS.")
+    for table in FERCEQR_TRANSFORM_ASSETS:
+        logger.info(f"Copying {table} to GCS.")
+        base_path = UPath(output_location) / table
+        base_path.mkdir(exist_ok=True)
+
+        # Loop through partitioned parquet files for table and write to GCS
+        for file in ParquetData(table_name=table).parquet_directory.iterdir():
+            destination_path = base_path / file.name
+            destination_path.write_bytes(file.read_bytes())
+
+    # Send slack notification about successful build
+    logger.info("Notifying slack about successful build.")
+    _notify_slack_deployments_channel(
+        ":large_green_circle: :sunglasses: :unicorn_face: :rainbow: ferceqr deployment succeeded!!"
+        " :partygritty: :database_parrot: :blob-dance: :large_green_circle:\n\n"
+        f"Parquet files can be found at: {output_location}"
+    )
+    _write_status_file("SUCCESS")
+
+
+@dg.asset
+def handle_ferceqr_deployment_failure():
+    """Send notification if EQR deployment failed."""
+    logger.info("Build failed, notifying slack.")
+    _notify_slack_deployments_channel(
+        message=":x: ferceqr deployment failed! See step status here:",
+        attached_file_path=str(_get_etl_status_csv_path()),
+    )
+    _write_status_file("FAILURE")
+
+
+def _get_etl_status_csv_path() -> Path:
+    return PudlPaths().output_dir / "ferceqr_etl_status.csv"
+
+
+@dg.sensor(
```

Comment:
This sensor runs every 60 seconds and will query the status of all the EQR assets to see if all partitions have been run, and if there were any failures. Once it detects that the job has completed, it will execute an asset to either handle a successful job or a failed one. I think the logic in this sensor could easily be generalized so we can apply this pattern to PUDL/SEC10k.